tests/integration/test_data_store_mgr.py

```diff
@@ -13,22 +13,103 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from copy import copy, deepcopy
 import pytest
 
+from cylc.flow import ID_DELIM
 from cylc.flow.data_store_mgr import (
     FAMILY_PROXIES,
+    JOBS,
     TASKS,
     TASK_PROXIES,
-    WORKFLOW
+    WORKFLOW,
+    JOB_STATUSES_ALL
+)
+from cylc.flow.task_state import (
+    TASK_STATUS_FAILED,
+    TASK_STATUS_SUCCEEDED,
+    TASK_STATUS_WAITING
 )
+from cylc.flow.wallclock import get_current_time_string
+
+
+def job_config(schd):
+    return {
+        'owner': schd.owner,
+        'host': 'commet',
+        'submit_num': 3,
+        'task_id': 'foo.1',
+        'job_runner_name': 'background',
+        'env-script': None,
+        'err-script': None,
+        'exit-script': None,
+        'execution_time_limit': None,
+        'init-script': None,
+        'post-script': None,
+        'pre-script': None,
+        'script': 'sleep 5; echo "I come in peace"',
+        'work_d': None,
+        'directives': {},
+        'environment': {},
+        'param_var': {},
+        'logfiles': [],
+        'platform': {'name': 'platform'},
+    }
+
+
+@pytest.fixture
+def job_db_row():
+    return [
+        '1',
+        'foo',
+        'running',
+        4,
+        '2020-04-03T13:40:18+13:00',
+        '2020-04-03T13:40:20+13:00',
+        '2020-04-03T13:40:30+13:00',
+        'background',
+        '20542',
+        'localhost',
+    ]
+
+
+@pytest.fixture
+@pytest.mark.asyncio
+async def myflow(flow, scheduler):
+    schd = scheduler(reg)
+    await schd.install()
+    await schd.initialise()
+    await schd.configure()
+    return schd
+
+
+def ext_id(schd):
+    return (
+        f'{schd.owner}{ID_DELIM}{schd.suite}{ID_DELIM}'
+        f'1{ID_DELIM}foo{ID_DELIM}3'
+    )
+
+
+def int_id(_):
+    return '1/foo/03'
 
 
 @pytest.mark.asyncio
 @pytest.fixture(scope='module')
-async def harness(mod_flow, mod_scheduler, mod_run, mod_one_conf):
-    reg = mod_flow(mod_one_conf)
+async def harness(mod_flow, mod_scheduler, mod_run):
+    flow_def = {
+        'scheduling': {
+            'graph': {
+                'R1': 'foo => bar'
+            }
+        }
+    }
+    reg = mod_flow(flow_def)
     schd = mod_scheduler(reg)
     async with mod_run(schd):
+        # Think this is needed to save the data state at first start (?)
+        # Fails without it.. and a test needs to overwrite schd data with this.
```

Comment:
Yeah, because this is a module-scoped fixture it is the same scheduler object provided to each test. When you do a `TaskProxy.reset` in one test that is messing with the object before it is passed to the next test.

You could potentially bolt a clear method onto the scheduler object which does something like `TaskProxy.reset(TASK_STATUS_WAITING, is_held=False)` to wipe the slate called something like `schd._test_clear()`.

Under this framework tests in a module are run in order, however, tests can be run manually one-at-a-time and have the potential to behave differently.